src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -345,17 +349,205 @@ bool Pane::ResizePane(const ResizeDirection& direction)
+        return;
+    }
+
+    const bool isFirstChild = sender == _firstChild;
+    // We want to handle this drag in the following cases
+    // * In a vertical split: if we're dragging the right of the first pane or the left of the second
+    // * In a horizontal split: if we're dragging the bottom of the first pane or the top of the second
+    const auto sideMatched = (_splitState == SplitState::Vertical)   ? (isFirstChild && WI_IsFlagSet(side, Borders::Right)) || (!isFirstChild && WI_IsFlagSet(side, Borders::Left)) :
+                             (_splitState == SplitState::Horizontal) ? (isFirstChild && WI_IsFlagSet(side, Borders::Bottom)) || (!isFirstChild && WI_IsFlagSet(side, Borders::Top)) :
+                                                                       false;
+
+    if (sideMatched)
+    {
+        _handleManipulation(delta);
+    }
+    else
+    {
+        // Bubble, with us as the new sender.
+        ManipulationRequested.raise(shared_from_this(), delta, side);
+    }
+}
+
+// Actually handle resizing our split in response to a drag event. If we're
+// being called, then we know that the delta that's passed to us should be
+// applied to our own split.  The delta that's passed in here is in PIXELS, not
+// DIPs.
+void Pane::_handleManipulation(const winrt::Windows::Foundation::Point delta)
+{
+    const auto scaleFactor = DisplayInformation::GetForCurrentView().RawPixelsPerViewPixel();
+
+    const auto weAreVertical = _splitState == SplitState::Vertical;
+    const winrt::Windows::Foundation::Point translationForUs = (weAreVertical) ? Point{ delta.X, 0 } : Point{ 0, delta.Y };
+
+    // Decide on direction based on delta
+    ResizeDirection dir = ResizeDirection::None;
+    if (_splitState == SplitState::Vertical)
+    {
+        if (translationForUs.X < 0)
+        {
+            dir = ResizeDirection::Left;
+        }
+        else if (translationForUs.X > 0)
+        {
+            dir = ResizeDirection::Right;
+        }
+    }
+    else if (_splitState == SplitState::Horizontal)
+    {
+        if (translationForUs.Y < 0)
+        {
+            dir = ResizeDirection::Up;
+        }
+        else if (translationForUs.Y > 0)
+        {
+            dir = ResizeDirection::Down;
+        }
+    }
+
+    // Resize in the given direction
+    if (dir != ResizeDirection::None)
+    {
+        // turn delta into a percentage
+        base::ClampedNumeric<float> amount;
+        base::ClampedNumeric<float> actualDimension;
+        if (dir == ResizeDirection::Left || dir == ResizeDirection::Right)
+        {
+            amount = translationForUs.X;
+            actualDimension = base::ClampedNumeric<float>(_root.ActualWidth());
+        }
+        else if (dir == ResizeDirection::Up || dir == ResizeDirection::Down)
+        {
+            amount = translationForUs.Y;
+            actualDimension = base::ClampedNumeric<float>(_root.ActualHeight());
+        }
+        const auto scaledAmount = amount * scaleFactor;

```

Comment:
Are you sure this needs to be scaled? Shouldn't the `delta` be in DIPs just like the ActualWidth/Height?

Reply:
YES LEONARD, IT SHOULD BE. It isn't, and I can only imagine it's because _~_ _XAML Islands reasons_ _~_

Reply:
Hmm I guess that's actually the intended behavior?

.NET seems to use DIPs (nice, consistent, and simple): https://learn.microsoft.com/en-us/dotnet/api/system.windows.frameworkelement.actualheight?view=windowsdesktop-8.0

WinUI seems to use "pixels" (wat, wat, and wat): https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.frameworkelement.actualheight?view=winrt-22621

Reply:
you know what the worst part is? I don't even think that documentation is correct. I'm pretty confident that `ActualWidth/Height` _is_ DIPs, and the `args.Delta().Translation` is ALSO _supposed_ to be DIPs, but in XAML islands, it's _not_. 

There was a similar issue like this some time ago: #2606 (which leads to MSFT:35087537)